### PR TITLE
Clang format 12

### PR DESCRIPTION
**Pull request by rafaeldtinoco**
## Description (git log)

commit e49dc500 (HEAD -> clang-format-12, rafaeldtinoco/clang-format-12)
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Sun Jan 15 04:38:15 2023

    builder: adjust OPA version to latest in remaining

commit 53acf4b0
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Sun Jan 15 04:30:12 2023

    builder: adjust tracee-make docker files
    
    - ubuntu: allow fix-fmt makefile rule to work inside tracee-make img
    - ubuntu: install multiple llvm/clang/clangd/clang-format versions
    
    - ubuntu & alpine: update opa to latest version to tracee-make img
    - ubuntu & alpine: add binaries for testing within tracee-make img

commit 06296884
Author: Rafael David Tinoco <rafaeldtinoco@gmail.com>
Date:   Sun Jan 15 04:28:48 2023

    git: update .gitignore
    
    - add temp files related to development

### CODE REVIEW

**Comment by rafaeldtinoco** [comment]
@yanivagman this fixes the clang-format-12 missing binary issue for the "tracee-make" ubuntu flavor. Now you're able to run "fix-fmt" inside the "ubuntu-shell", as you said.

Also, I've made some adjustments to the OPA version for these environments. I'm now fixing the OS packaging (it's a quick fix) for new fedora and ubuntu versions, as well as including single binary on them.

File: builder/Dockerfile.ubuntu-tracee-make [hunk 1]
```diff
@@ -11,19 +11,38 @@ FROM ubuntu:jammy
...
     apt-get install -y linux-headers-generic && \
     apt-get install -y libelf-dev && \
     apt-get install -y zlib1g-dev && \
-    curl -L -o /usr/bin/opa https://github.com/open-policy-agent/opa/releases/download/v0.44.0/opa_linux_amd64_static && \
+    update-alternatives --install /usr/bin/clang clang /usr/bin/clang-12 130 --slave /usr/bin/clang++ clang++ /usr/bin/clang++-12 --slave /usr/bin/llc llc /usr/bin/llc-12 --slave /usr/bin/clang-format clang-format /usr/bin/clang-format-12 --slave /usr/bin/clangd clangd /usr/bin/clangd-12 && \
+    update-alternatives --install /usr/bin/clang clang /usr/bin/clang-13 130 --slave /usr/bin/clang++ clang++ /usr/bin/clang++-13 --slave /usr/bin/llc llc /usr/bin/llc-13 --slave /usr/bin/clang-format clang-format /usr/bin/clang-format-13 --slave /usr/bin/clangd clangd /usr/bin/clangd-13 && \
+    update-alternatives --install /usr/bin/clang clang /usr/bin/clang-14 140 --slave /usr/bin/clang++ clang++ /usr/bin/clang++-14 --slave /usr/bin/llc llc /usr/bin/llc-14 --slave /usr/bin/clang-format clang-format /usr/bin/clang-format-14 --slave /usr/bin/clangd clangd /usr/bin/clangd-14 && \
```
Comment (yanivagman) [inline builder/Dockerfile.ubuntu-tracee-make:33]: DO we really need all clang 12-13-14 versions?

Comment (rafaeldtinoco) [reply builder/Dockerfile.ubuntu-tracee-make:33]: It makes it easier for us to check incompatibility issues (like the one PR: #2581 fixes). Especially for developers using rolling-based distros (Ubuntu keeps multiple toolchain versions but others distros don't). 

Also, this image is for "development" purposes and won't be added to any official published image (raising its size nor anything like it).

**Review by yanivagman — APPROVED** [review]
LGTM

**Pull request merged**